src/mappings/helpers.ts

```diff
@@ -55,21 +53,50 @@ export function updateCommonCTokenStats(
   marketID: string,
   marketSymbol: string,
   accountID: string,
-  txHash: Bytes,
-  timestamp: i32,
-  blockNumber: i32,
+  tx_hash: Bytes,
+  timestamp: BigInt,
+  blockNumber: BigInt,
+  logIndex: BigInt,
 ): AccountCToken {
   let cTokenStatsID = marketID.concat('-').concat(accountID)
   let cTokenStats = AccountCToken.load(cTokenStatsID)
   if (cTokenStats == null) {
     cTokenStats = createAccountCToken(cTokenStatsID, marketSymbol, accountID, marketID)
   }
-  let txHashes = cTokenStats.transactionHashes
-  txHashes.push(txHash)
-  cTokenStats.transactionHashes = txHashes
-  let txTimes = cTokenStats.transactionTimes
-  txTimes.push(timestamp)
-  cTokenStats.transactionTimes = txTimes
+  getOrCreateAccountCTokenTransaction(
+    cTokenStatsID,
+    tx_hash,
+    timestamp,
+    blockNumber,
+    logIndex,
+  )
   cTokenStats.accrualBlockNumber = blockNumber
   return cTokenStats as AccountCToken
 }
+
+export function getOrCreateAccountCTokenTransaction(
+  accountID: string,
+  tx_hash: Bytes,
+  timestamp: BigInt,
+  block: BigInt,
+  logIndex: BigInt,
+): AccountCTokenTransaction {
+  let id = accountID
+    .concat('-')
+    .concat(tx_hash.toHexString())
+    .concat('-')
+    .concat(logIndex.toString())
+  let transaction = AccountCTokenTransaction.load(id)
+
+  if (transaction == null) {
```

Comment:
My only comment would be, i believe this will always be true. The id is `txhash-logIndex` . Each tx is unique, and each log index is unique in a tx. So it might be better named `createAccountCTokenTransaction()` 

Do you agree here?

Reply:
Yeah, it should always be true, except in the odd case where you need to reconstruct the entity because of a refactor in the handlers (which really shouldn't happen unless we start tracking lots of data for that entity and want to separate the different concerns in different auxiliary functions).

I almost always leave the helpers with a common naming convention, even if they won't really be used to actually `get` an already existing entity most of the time, just for the sake of consistency across all helper creation functions.